magicctapipe/scripts/lst1_magic/semi_automatic_scripts/coincident_events.py

```diff
@@ -0,0 +1,374 @@
+                            f"export OUTPUTDIR={outputdir}\n",
+                            "SAMPLE_LIST=($(<$OUTPUTDIR/logs/list_LST.txt))\n",
+                            "SAMPLE=${SAMPLE_LIST[${SLURM_ARRAY_TASK_ID}]}\n",
+                            "export LOG=$OUTPUTDIR/logs/coincidence_${SLURM_ARRAY_TASK_ID}.log\n",
+                            f"time conda run -n {env_name} lst1_magic_event_coincidence --input-file-lst $SAMPLE --input-dir-magic $INM --output-dir $OUTPUTDIR --config-file {target_dir}/config_coincidence.yaml >$LOG 2>&1",
+                        ]
+                        with open(
+                            f"{source}_LST_coincident_{nsb}_{outputdir.split('/')[-1]}.sh",
+                            "w",
+                        ) as f:
+                            f.writelines(lines)
+    else:
+        coincidence_DL1_dir = f"{target_dir}/DL1/Observations"
+        if not os.path.exists(f"{coincidence_DL1_dir}/Coincident"):
+            os.mkdir(f"{coincidence_DL1_dir}/Coincident")
+
+        for i in LST_runs:
+            lstObsDir = i[0].split("_")[0] + i[0].split("_")[1] + i[0].split("_")[2]
+            inputdir = f"/fefs/aswg/data/real/DL1/{lstObsDir}/{LST_version}/tailcut84"
+            outputdir = f"{coincidence_DL1_dir}/Coincident/{lstObsDir}"
+            list_of_subruns = np.sort(glob.glob(f"{inputdir}/dl1*Run*{i[1]}*.*.h5"))
+            if os.path.exists(f"{outputdir}/list_LST.txt"):
+                with open(f"{outputdir}/list_LST.txt", "a") as LSTdataPathFile:
+                    for subrun in list_of_subruns:
+                        LSTdataPathFile.write(
+                            f"{subrun}\n"
+                        )  # If this files already exists, simply append the new information
+            else:
+                os.mkdir(outputdir)
+                with open(
+                    f"{outputdir}/list_LST.txt", "w"
+                ) as f:  # If the file list_LST.txt does not exist, it will be created here
+                    for subrun in list_of_subruns:
+                        f.write(f"{subrun}\n")
+        process_name = target_dir.split("/")[-2:][1]
+
+        listOfNightsLST = np.sort(
+            glob.glob(f"{target_dir}/DL1/Observations/Coincident/*")
+        )
+        listOfNightsMAGIC = np.sort(
+            glob.glob(f"{target_dir}/DL1/Observations/Merged/Merged*")
+        )
+
+        for nightMAGIC, nightLST in zip(listOfNightsMAGIC, listOfNightsLST):
+            process_size = (
+                len(np.genfromtxt(f"{nightLST}/list_LST.txt", dtype="str")) - 1
+            )
+
+            with open(f"LST_coincident_{nightLST.split('/')[-1]}.sh", "w") as f:
+                lines = [
+                    "#!/bin/sh\n\n",
+                    "#SBATCH -p short\n",
+                    f"#SBATCH -J {process_name}_coincidence\n",
+                    f"#SBATCH --array=0-{process_size}%50\n",
```

Comment:
why do you limit the number of jobs to 50? I would not do so and let the cluster assign the priorities. This way if the cluster is empty at a moment it can use its whole power.